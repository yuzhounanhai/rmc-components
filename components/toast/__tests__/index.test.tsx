import React from 'react';
import { act } from 'react-dom/test-utils';
import CheckOutlined from '@ant-design/icons/lib/icons/CheckOutlined'
import { mount } from 'enzyme';
import FadeIn from '../../fade/fadeIn';
import baseTest from '../../../tests/common/baseTest';
import Toast from '..';
import BaseToast from '../toast';
import { sleep } from '../../../tests/common/utils';

describe('Toast', () => {
  baseTest(
    <BaseToast
      content="content"
    />
  );

  baseTest(
    <BaseToast
      content="content"
      smooth
    />
  );

  baseTest(
    <BaseToast
      content="content"
      round
    />
  );

  baseTest(
    <BaseToast
      content="content"
      at="top"
    />
  );

  baseTest(
    <BaseToast
      content="content"
      at="bottom"
    />
  );

  baseTest(
    <BaseToast
      content="content"
      icon={(
        <CheckOutlined />
      )}
    />
  );

  it('onHide callback should trigger when Toast is hide', async () => {
    const onHideCb = jest.fn();
    const wrapper = mount(
      <Toast
        content="content"
        onHide={onHideCb}
      />
    );
    await act(async () => {
      await sleep(50);
      wrapper.update();
    });
    wrapper.find(FadeIn).simulate('transitionend');
    await act(async () => {
      await sleep(3000);
      wrapper.update();
    });
    expect(wrapper.find(FadeIn).prop('show')).toBeFalsy();
    wrapper.find(FadeIn).simulate('transitionend');
    expect(onHideCb).toBeCalled();
  });

  it('"duration" prop should control the Toast components time of show.', async () => {
    const onHideCb = jest.fn();
    const wrapper = mount(
      <Toast
        content="content"
        onHide={onHideCb}
        duration={3000}
      />
    );
    await act(async () => {
      await sleep(50);
      wrapper.update();
    });
    wrapper.find(FadeIn).simulate('transitionend');
    await act(async () => {
      await sleep(2000);
      wrapper.update();
    });
    expect(wrapper.find(FadeIn).prop('show')).toBeTruthy();
    await act(async () => {
      await sleep(1000);
      wrapper.update();
    });
    expect(wrapper.find(FadeIn).prop('show')).toBeFalsy();
    expect(onHideCb).not.toBeCalled();
    wrapper.find(FadeIn).simulate('transitionend');
    expect(onHideCb).toBeCalled();
  });

  it('"transitionSpeed" props need give to <FadeIn/>', () => {
    const speed = 1;
    const wrapper = mount(
      <Toast
        content="content"
        transitionSpeed={speed}
      />
    );
    expect(wrapper.find(FadeIn).prop('speed')).toBe(speed);
  });

  it('quick function to create Toast should be defined.', () => {
    expect(Toast.info).toBeDefined();
    expect(Toast.loading).toBeDefined();
    expect(Toast.delayLoading).toBeDefined();
    expect(Toast.fail).toBeDefined();
    expect(Toast.success).toBeDefined();
    expect(Toast.hide).toBeDefined();
  });

  it('Call "Toast.hide" should not throw any warning or error where Toast is not exist.', () => {
    expect(() => {
      Toast.hide();
    }).not.toThrow();
  });
});