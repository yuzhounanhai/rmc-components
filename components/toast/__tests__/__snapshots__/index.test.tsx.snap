// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast component snapshots tests 1`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "rmcc-toast",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "rmcc-toast-content rmcc-toast-smooth rmcc-fade rmcc-fade-op-0",
          "style": "transition-property:opacity;transition-duration:0.5s;transition-timing-function:ease",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "rmcc-toast-text",
            },
            "children": Array [
              Node {
                "data": "content",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Toast component snapshots tests 2`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "rmcc-toast",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "rmcc-toast-content rmcc-toast-smooth rmcc-fade rmcc-fade-op-0",
          "style": "transition-property:opacity;transition-duration:0.5s;transition-timing-function:ease",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "rmcc-toast-text",
            },
            "children": Array [
              Node {
                "data": "content",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Toast component snapshots tests 3`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "rmcc-toast",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "rmcc-toast-content rmcc-toast-round rmcc-toast-smooth rmcc-fade rmcc-fade-op-0",
          "style": "transition-property:opacity;transition-duration:0.5s;transition-timing-function:ease",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "rmcc-toast-text",
            },
            "children": Array [
              Node {
                "data": "content",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Toast component snapshots tests 4`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "rmcc-toast rmcc-toast-at-top",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "rmcc-toast-content rmcc-toast-smooth rmcc-fade rmcc-fade-op-0",
          "style": "transition-property:opacity;transition-duration:0.5s;transition-timing-function:ease",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "rmcc-toast-text",
            },
            "children": Array [
              Node {
                "data": "content",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Toast component snapshots tests 5`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "rmcc-toast rmcc-toast-at-bottom",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "rmcc-toast-content rmcc-toast-smooth rmcc-fade rmcc-fade-op-0",
          "style": "transition-property:opacity;transition-duration:0.5s;transition-timing-function:ease",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "rmcc-toast-text",
            },
            "children": Array [
              Node {
                "data": "content",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Toast component snapshots tests 6`] = `
initialize {
  "0": Node {
    "attribs": Object {
      "class": "rmcc-toast",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "rmcc-toast-content rmcc-toast-smooth rmcc-toast-has-icon rmcc-fade rmcc-fade-op-0",
          "style": "transition-property:opacity;transition-duration:0.5s;transition-timing-function:ease",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "aria-label": "check",
              "class": "anticon anticon-check",
              "role": "img",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "aria-hidden": "true",
                  "data-icon": "check",
                  "fill": "currentColor",
                  "focusable": "false",
                  "height": "1em",
                  "viewBox": "64 64 896 896",
                  "width": "1em",
                },
                "children": Array [
                  Node {
                    "attribs": Object {
                      "d": "M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 00-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z",
                    },
                    "children": Array [],
                    "name": "path",
                    "namespace": "http://www.w3.org/2000/svg",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "d": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "d": undefined,
                    },
                  },
                ],
                "name": "svg",
                "namespace": "http://www.w3.org/2000/svg",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "data-icon": undefined,
                  "fill": undefined,
                  "focusable": undefined,
                  "height": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "data-icon": undefined,
                  "fill": undefined,
                  "focusable": undefined,
                  "height": undefined,
                  "viewBox": undefined,
                  "width": undefined,
                },
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": Node {
              "attribs": Object {
                "class": "rmcc-toast-text",
              },
              "children": Array [
                Node {
                  "data": "content",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "text",
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-label": undefined,
              "class": undefined,
              "role": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-label": undefined,
              "class": undefined,
              "role": undefined,
            },
          },
          Node {
            "attribs": Object {
              "class": "rmcc-toast-text",
            },
            "children": Array [
              Node {
                "data": "content",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "text",
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": Node {
              "attribs": Object {
                "aria-label": "check",
                "class": "anticon anticon-check",
                "role": "img",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "data-icon": "check",
                    "fill": "currentColor",
                    "focusable": "false",
                    "height": "1em",
                    "viewBox": "64 64 896 896",
                    "width": "1em",
                  },
                  "children": Array [
                    Node {
                      "attribs": Object {
                        "d": "M912 190h-69.9c-9.8 0-19.1 4.5-25.1 12.2L404.7 724.5 207 474a32 32 0 00-25.1-12.2H112c-6.7 0-10.4 7.7-6.3 12.9l273.9 347c12.8 16.2 37.4 16.2 50.3 0l488.4-618.9c4.1-5.1.4-12.8-6.3-12.8z",
                      },
                      "children": Array [],
                      "name": "path",
                      "namespace": "http://www.w3.org/2000/svg",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "d": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "d": undefined,
                      },
                    },
                  ],
                  "name": "svg",
                  "namespace": "http://www.w3.org/2000/svg",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "data-icon": undefined,
                    "fill": undefined,
                    "focusable": undefined,
                    "height": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "data-icon": undefined,
                    "fill": undefined,
                    "focusable": undefined,
                    "height": undefined,
                    "viewBox": undefined,
                    "width": undefined,
                  },
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-label": undefined,
                "class": undefined,
                "role": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-label": undefined,
                "class": undefined,
                "role": undefined,
              },
            },
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
